src/main/java/org/jboss/tools/intellij/openshift/utils/odo/JSonParser.java

```diff
@@ -12,251 +12,241 @@
-        ComponentInfo.Builder builder = new ComponentInfo.Builder().withComponentKind(ComponentKind.DEVFILE);
-        if (root.has(PROJECT_TYPE_FIELD)) {
-            String componentTypeName = root.get(PROJECT_TYPE_FIELD).asText();
-            builder.withComponentTypeName(componentTypeName);
-        }
-        return builder.build();
-    }
-
-    public ComponentInfo parseDescribeComponentInfo(ComponentKind kind, boolean isPodmanPresent) {
-        ComponentInfo.Builder builder = new ComponentInfo.Builder().withComponentKind(kind);
-        if (root.has(DEVFILE_DATA_FIELD) && root.get(DEVFILE_DATA_FIELD).has(DEVFILE_FIELD) &&
-            root.get(DEVFILE_DATA_FIELD).get(DEVFILE_FIELD).has(METADATA_FIELD) &&
-            root.get(DEVFILE_DATA_FIELD).get(DEVFILE_FIELD).get(METADATA_FIELD).has(PROJECT_TYPE_FIELD)) {
-            String componentTypeName = root.get(DEVFILE_DATA_FIELD).get(DEVFILE_FIELD).get(METADATA_FIELD).get(PROJECT_TYPE_FIELD).asText();
-            builder.withComponentTypeName(componentTypeName);
-        }
-        if (root.has(DEVFILE_DATA_FIELD) && root.get(DEVFILE_DATA_FIELD).has(DEVFILE_FIELD) &&
-            root.get(DEVFILE_DATA_FIELD).get(DEVFILE_FIELD).has(METADATA_FIELD) &&
-            root.get(DEVFILE_DATA_FIELD).get(DEVFILE_FIELD).get(METADATA_FIELD).has(LANGUAGE_FIELD)) {
-            String language = root.get(DEVFILE_DATA_FIELD).get(DEVFILE_FIELD).get(METADATA_FIELD).get(LANGUAGE_FIELD).asText();
-            builder.withLanguage(language);
-        }
-        if (root.has(DEVFILE_DATA_FIELD) && root.get(DEVFILE_DATA_FIELD).has(SUPPORTED_ODO_FEATURES_FIELD)) {
-            JsonNode featuresNode = root.get(DEVFILE_DATA_FIELD).get(SUPPORTED_ODO_FEATURES_FIELD);
-            List<ComponentFeature.Mode> features = getComponentsFeatures(featuresNode);
-            builder.withSupportedFeatures(features);
-        }
-         return builder.withLocalPodmanPresence(isPodmanPresent).build();
-    }
-
-    private List<ComponentFeature.Mode> getComponentsFeatures(JsonNode featuresNode) {
-        List<ComponentFeature.Mode> result = new ArrayList<>();
-        if (featuresNode.has(ComponentFeature.Mode.DEV_MODE.getLabel().toLowerCase()) && featuresNode.get(ComponentFeature.Mode.DEV_MODE.getLabel().toLowerCase()).asBoolean()) {
-            result.add(ComponentFeature.Mode.DEV_MODE);
-        }
-        if (featuresNode.has(ComponentFeature.Mode.DEBUG_MODE.getLabel().toLowerCase()) && featuresNode.get(ComponentFeature.Mode.DEBUG_MODE.getLabel().toLowerCase()).asBoolean()) {
-            result.add(ComponentFeature.Mode.DEBUG_MODE);
-        }
-        if (featuresNode.has(ComponentFeature.Mode.DEPLOY_MODE.getLabel().toLowerCase()) && featuresNode.get(ComponentFeature.Mode.DEPLOY_MODE.getLabel().toLowerCase()).asBoolean()) {
-            result.add(ComponentFeature.Mode.DEPLOY_MODE);
-        }
-        return result;
-    }
-
-    public DebugStatus parseDebugStatus() {
```

Comment:
Afaics this method was never used and could therefore be eliminated +1. Afaiu this has the benefit that apache commons is not used any more.

Reply:
i'm looking into other places that *still* uses apache Stringutils class